src/http.ts

```diff
@@ -260,15 +260,30 @@ function httpRequest<TData, TError, TTransform>(url: string, options?: RequestIn
 				}
 				throw error;
 			}
-			const text = await response.text();
-			const isJson = !!text && (text[0] === '{' || text[0] === '[');
+
 			let data: TData;
-			try {
-				// eslint-disable-next-line @typescript-eslint/no-explicit-any
-				data = (isJson ? JSON.parse(text) : (text as any)) as TData;
-			} catch {
-				// eslint-disable-next-line @typescript-eslint/no-explicit-any
-				data = (text as any) as TData;
+			const contentTypes = response.headers.get('content-type')?.split(',');
+			if (
+				contentTypes?.includes(ContentTypes.Bmp) ||
+				contentTypes?.includes(ContentTypes.Gif) ||
+				contentTypes?.includes(ContentTypes.Jpeg) ||
+				contentTypes?.includes(ContentTypes.Pdf) ||
+				contentTypes?.includes(ContentTypes.Png) ||
+				contentTypes?.includes(ContentTypes.Gzip) ||
+				contentTypes?.includes(ContentTypes.OctetStream) ||
+				contentTypes?.includes(ContentTypes.Zip)
+			) {
+				data = ((await response.blob()) as unknown) as TData;
+			} else {
+				const text = await response.text();
+				const isJson = !!text && (text[0] === '{' || text[0] === '[');
+				try {
+					// eslint-disable-next-line @typescript-eslint/no-explicit-any
+					data = (isJson ? JSON.parse(text) : (text as any)) as TData;
```

Comment:
This actually handles JSON better than the built in json() with no error handling :+1: